From f524834fd97006b3bcdee919ab783f20c231dda9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jos=C3=A9=20Exp=C3=B3sito?= <jose.exposito89@gmail.com>
Date: Mon, 9 Feb 2026 10:02:18 +0100
Subject: [PATCH 2/2] venus: Fix error log on PPC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On the ppc64le architecture error log fail to compile with error:

    ../src/virtio/vulkan/vn_renderer_virtgpu.c: In function ‘virtgpu_ioctl_map’:
    ../src/virtio/vulkan/vn_renderer_virtgpu.c:751:66: error: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 6 has type ‘__u64’ {aka ‘long unsigned int’} [-Werror=format=]
    751 |          "mmap failed: gpu_fd=%d, handle=%u, size=%zu, offset=%llu, err=%s",
        |                                                               ~~~^
        |                                                                  |
        |                                                                  long long unsigned int
        |                                                               %lu
    752 |          gpu->fd, gem_handle, size, args.offset, strerror(errno));
        |                                     ~~~~~~~~~~~
        |                                         |
        |                                         __u64 {aka long unsigned int}
    cc1: some warnings being treated as errors

Parse the parameters to fix the failure.

Fixes: a49b7adad883 ("venus: add error log coverage for virtgpu backend")
---
 src/virtio/vulkan/vn_renderer_virtgpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/virtio/vulkan/vn_renderer_virtgpu.c b/src/virtio/vulkan/vn_renderer_virtgpu.c
index 575ab9cc4fe..9e233b560d3 100644
--- a/src/virtio/vulkan/vn_renderer_virtgpu.c
+++ b/src/virtio/vulkan/vn_renderer_virtgpu.c
@@ -749,7 +749,7 @@ virtgpu_ioctl_map(struct virtgpu *gpu,
       vn_log(
          gpu->instance,
          "mmap failed: gpu_fd=%d, handle=%u, size=%zu, offset=%llu, err=%s",
-         gpu->fd, gem_handle, size, args.offset, strerror(errno));
+         gpu->fd, gem_handle, size, (long long)args.offset, strerror(errno));
       return NULL;
    }
 
-- 
2.53.0

